testng-core/src/test/java/org/testng/JarFileUtilsTest.java

```diff
@@ -89,11 +126,32 @@ public void testWithInvalidXmlFile() throws MalformedURLException {
    */
   @Test
   public void ensureThatExceptionAreNotThrown() throws MalformedURLException {
+    TestNG testNg = new TestNG(false);
+    List<String> testNames =
+        Arrays.asList("testng-tests-child2", "testng-tests-child4", "testng-tests-child5");
+    testNg.setTestNames(testNames);
+    testNg.setXmlPathInJar("jarfileutils/testng-tests.xml");
+    testNg.setTestJar(jar.getAbsolutePath());
+    testNg.initializeSuitesAndJarFile();
+    // "testng-tests-child2", "testng-tests-child4", "testng-tests-child5" are from 3 different test
+    // suites
+    Assert.assertEquals(testNg.m_suites.size(), 3);
+  }
+
+  /**
+   * Test to ensure that exception is thrown for invalid test name.
+   *
+   * @throws MalformedURLException
+   */
+  @Test(
+      expectedExceptions = TestNGException.class,
+      expectedExceptionsMessageRegExp = "\nThe test\\(s\\) <\\[dummy\\]> cannot be found in suite.")
+  public void ensureThatExceptionAreThrown() throws MalformedURLException {
     TestNG testNg = new TestNG(false);
     List<String> testNames =
         Arrays.asList("testng-tests-child2", "testng-tests-child4", "testng-tests-child5", "dummy");
     testNg.setTestNames(testNames);
-    testNg.setXmlPathInJar(jar.getAbsolutePath());
+    testNg.setXmlPathInJar("jarfileutils/testng-tests.xml");
```

Comment:
This is the fix for legacy test case issue: used wrong value for xml path in jar.